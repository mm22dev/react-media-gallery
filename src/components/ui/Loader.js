import React from 'react'
import PropTypes from 'prop-types'
import { connectWithPopupSettings } from '../../providers/PopupSettings'
import { Fade } from '../_common'
import { pixelify } from '../../lib/helpers'

const LoaderUI = ({ style, animate, fade, innerMarginTop, popupSettings }) => {
  const { usePopupDefaultStyling, popupLoaderSelector, popupLoaderTextSize, popupLoaderText, popupTextColor, popupHeight } = popupSettings

  const defaultWrapperStyle = { ...(usePopupDefaultStyling && {
    position: 'relative',
    fontSize: popupLoaderTextSize
  }) }

  const defaultContentStyle = { ...(usePopupDefaultStyling && {
    height: `${Math.floor(popupHeight / 2)}px`,
    overflow: 'hidden',
    lineHeight: `${Math.floor(popupHeight / 2)}px`,
    textAlign: 'center',
    color: popupTextColor || '',
    ...(animate && innerMarginTop !== 0 && { marginTop: pixelify(innerMarginTop) })
  }) }

  return (
    // PopupSettings provider sets popupLoaderSelector to null if usePopupLoader is false
    popupLoaderSelector
      ? <Fade {...fade} customDisplayOnFadeOutEnd={'none'}>
        <div className={`prt-l-default ${popupLoaderSelector.slice(1)}`} style={{ ...defaultWrapperStyle, ...style }}>
          <div style={defaultContentStyle} className={animate && usePopupDefaultStyling ? 'animated' : 'not-animated'}>{
            popupLoaderText
          }</div>
        </div>
      </Fade>
      : null
  )
}

LoaderUI.propTypes = {
  style: PropTypes.object,
  animate: PropTypes.bool,
  fade: PropTypes.object,
  innerMarginTop: PropTypes.number,
  popupSettings: PropTypes.object
}

LoaderUI.defaultProps = {
  style: {},
  animate: false,
  fade: { inProp: false, duration: 0, immediate: true },
  innerMarginTop: 0
}

const Loader = connectWithPopupSettings(LoaderUI)
Loader.displayName = 'Loader'
export default Loader
