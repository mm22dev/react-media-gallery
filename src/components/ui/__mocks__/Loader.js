const Loader = () => <div className='loader loader-mock'>
  <div className='loader-content-mock'></div>
</div>

Loader.displayName = 'LoaderMock'

export default Loader
