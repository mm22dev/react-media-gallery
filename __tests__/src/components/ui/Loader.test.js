import Loader from '../../../../src/components/ui/Loader'
import toJSON from 'enzyme-to-json'
import { compose } from 'redux'
import defaultPopupSettings from '../../../../src/data/popupSettings.json'

const { shallow, mount } = Enzyme

describe('<Loader /> Component', () => {
  let { popupSettings } = defaultPopupSettings
  let wrapper

  describe('Rendering UI', () => {
    it('Renders correctly', () => {
      compose(expect, toJSON, shallow)(<Loader />).toMatchSnapshot()
    })

    it('Renders empty Component if required', () => {
      expect(mount(<Loader popupSettings={{ ...popupSettings, usePopupLoader: false, popupLoaderSelector: false }} />).isEmptyRender()).toBe(true)
    })

    it('Does not Apply default style if required', () => {
      const fadeInjectedStyle = { display: 'none', opacity: 0 }
      const parentInjectedStyle = {}
      const defaultWrapperStyle = {}
      wrapper = mount(<Loader popupSettings={{ ...popupSettings, usePopupDefaultStyling: false }} />)
      expect(wrapper.find(popupSettings.popupLoaderSelector).get(0).props.style)
        .toEqual({
          ...fadeInjectedStyle,
          ...parentInjectedStyle,
          ...defaultWrapperStyle
        })
    })

    it('Triggers CSS animation on loading request when default Popup style is applied', () => {
      wrapper = mount(<Loader popupSettings={{ ...popupSettings, usePopupDefaultStyling: true }} animate={true}/>)
      expect(wrapper.find(`${popupSettings.popupLoaderSelector} div`).get(0).props.className).toEqual(expect.stringContaining('animated'))
    })

    it('Renders a custom Loader text', () => {
      const _loaderText = 'mockedText'
      wrapper = mount(<Loader popupSettings={{ ...popupSettings, popupTextColor: null, popupLoaderText: _loaderText }} />)
      expect(wrapper.find('div').first().text()).toBe(_loaderText)
    })

    describe('Using Popup default style', () => {
      beforeAll(() => {
        wrapper = mount(<Loader popupSettings={{ ...popupSettings, usePopupDefaultStyling: true }} animate={true}/>)
      })

      it('Applies further margin top to content on animate request ', () => {
        expect(wrapper.find(`${popupSettings.popupLoaderSelector} div`).get(0).props.style.marginTop).not.toBeDefined()
        wrapper.setProps({ innerMarginTop: 100 })
        expect(wrapper.find(`${popupSettings.popupLoaderSelector} div`).get(0).props.style.marginTop).toBe('100px')
      })
    })
  })
})
