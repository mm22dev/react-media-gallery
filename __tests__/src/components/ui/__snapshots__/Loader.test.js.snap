// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Component Rendering UI Renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
